apps/server/src/tasks/entities/chunkInfo.entity.ts

```diff
@@ -1,33 +1,81 @@
-import { Column, Entity, PrimaryGeneratedColumn } from 'typeorm';
+import {
+  Column,
+  Entity,
+  JoinColumn,
+  OneToOne,
+  PrimaryGeneratedColumn,
+} from 'typeorm';
+import moment, { Duration } from 'moment';
 import { Field, ObjectType } from '@nestjs/graphql';
-import { Duration } from 'moment';
 import { IntervalColumn } from '../../types/IntervalColumn';
 import { GeneralEntity } from '../../utils/GeneralEntity';
+import { NullableColumn } from '../../utils/NullableColumn';
+import { NullableField } from '../../utils/NullableField';
 import { Interval } from '../../utils/interval.scalar';
+import { Repeat } from './repeat.entity';
 
+// TODO Those fields need to be virtual and better-organized.
 @Entity()
 @ObjectType()
 export class ChunkInfo extends GeneralEntity {
   @Field()
   @PrimaryGeneratedColumn()
   id: number;
 
+  // Shared properties
   @Field({
     description:
-      'The time when task should start. This can be different from taskBreakdown.start, as it is just informative data unrelated with real planed entity.',
+      'The time when task should start. In case of float tasks this can be different from taskBreakdown.start, as it is just informative data unrelated with real planed entity.',
   })
   @Column({ type: 'timestamp with time zone' })
   start: Date;
 
-  @Field(() => Interval)
+  @Field(() => Interval, {
+    // 15 minutes is a default value
+    defaultValue: moment.duration(15, 'minutes').toISOString(),
+    description:
+      'A minimum time gap that user wants to have between this task and another tasks.',
+  })
   @IntervalColumn()
-  minTimeBetweenChunks: Duration;
+  chillTime: Duration;
 
-  @Field(() => Interval)
-  @IntervalColumn()
+  // Const tasks properties
+
+  @NullableField(() => Repeat, {
+    description:
+      'Only to const tasks. Describes how often the task should repeat. When representing task in time, the tasks WILL be duplicated for the sake of easier calculations.',
+  })
+  @OneToOne(() => Repeat)
+  @JoinColumn()
+  repeat?: Repeat;
```

Comment:
is the `?` operator needed when there is `{ nullable: true }/@NullableField/@NullableColumn` decorator? Can we unify this somehow?

Reply:
I think yes, it's needed for ts to not shout at us when creating an instance of that entity.

Reply:
I don't think there is a way to unify those things. As they are responsible for completely different app parts.

Do you have any ideas on how we could unify that?